lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -163,6 +165,35 @@ def add_invoice(post, money, options)
         post[:metadata][:udf5] = application_id || 'ActiveMerchant'
       end
 
+      def add_recipient_data(post, options)
+        return unless options[:recipient] && options[:recipient].is_a?(Hash)
+
+        recipient = options[:recipient]
+
+        post[:recipient] = {}
+        post[:recipient][:dob] = recipient[:dob] if recipient[:dob]
+        post[:recipient][:zip] = recipient[:zip] if recipient[:zip]
+        post[:recipient][:account_number] = recipient[:account_number] if recipient[:account_number]
+        post[:recipient][:first_name] = recipient[:first_name] if recipient[:first_name]
+        post[:recipient][:last_name] = recipient[:last_name] if recipient[:last_name]
+
+        if address = recipient[:address]
+          post[:recipient][:address] = {}
+          post[:recipient][:address][:address_line1] = address[:address_line_1] if address[:address_line_1]
```

Comment:
This is nitpicky, but I think I'd prefer for us to expect `address_line1` instead of `address_line_1`

Since we are more or less mapping this directly to Checkout's hash structure, sneaking in that additional underscore could be confusing; it would be more clear if we match the Checkout naming conventions exactly